shell/platform/common/cpp/accessibility_bridge.h

```diff
@@ -0,0 +1,266 @@
+  ///             update. Calling this method alone will NOT update the
+  ///             semantics tree. To flush the pending updates, call the
+  ///             CommitUpdates().
+  ///
+  /// @param[in]  action           A pointer to the custom semantics action
+  /// update.
+  void AddFlutterSemanticsCustomActionUpdate(
+      const FlutterSemanticsCustomAction* action);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Flushes the pending updates and applies them to this
+  ///             accessibility bridge. Calling this with no pending updates
+  ///             does nothing, and callers should call this method at the end
+  ///             of an automic batch to avoid leaving the tree in a unstable
+  ///             state. For example if a node reparents from A to B, callers
+  ///             should only call this method when both removal from A and
+  ///             addition to B are in the pending updates.
+  void CommitUpdates();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the flutter platform node delegate with the given id from
+  ///             this accessibility bridge.
+  ///
+  /// @param[in]  id           The id of the flutter accessibility node you want
+  ///                          to retrieve.
+  std::weak_ptr<FlutterPlatformNodeDelegate>
+  GetFlutterPlatformNodeDelegateFromID(ui::AXNode::AXID id) const;
+
+  //------------------------------------------------------------------------------
+  /// @brief      Update the currently focused flutter accessibility node.
+  ///
+  /// @param[in]  id           The id of the currently focused flutter
+  ///                          accessibility node.
+  void SetFocusedNode(ui::AXNode::AXID node_id);
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the last focused node.
+  ui::AXNode::AXID GetLastFocusedNode();
+
+  //------------------------------------------------------------------------------
+  /// @brief      Get the ax tree data.
+  const ui::AXTreeData& GetAXTreeData() const;
+
+  // ui::AXTreeObserver implementation.
+  void OnNodeWillBeDeleted(ui::AXTree* tree, ui::AXNode* node) override;
+  void OnSubtreeWillBeDeleted(ui::AXTree* tree, ui::AXNode* node) override;
+  void OnNodeCreated(ui::AXTree* tree, ui::AXNode* node) override;
+  void OnNodeDeleted(ui::AXTree* tree, ui::AXNode::AXID node_id) override;
+  void OnNodeReparented(ui::AXTree* tree, ui::AXNode* node) override;
+  void OnRoleChanged(ui::AXTree* tree,
+                     ui::AXNode* node,
+                     ax::mojom::Role old_role,
+                     ax::mojom::Role new_role) override;
+  void OnAtomicUpdateFinished(
+      ui::AXTree* tree,
+      bool root_changed,
+      const std::vector<ui::AXTreeObserver::Change>& changes) override;
```

Comment:
Shouldn't these be private?


Also, the convention in the engine C++ code is to write the comment as something like

```cpp
// |ui::AxTreeObserver|
```